File: BOJ/C++/1026-보물.cpp
#include <iostream>
#include <algorithm>

bool calc(int a, int b)
{
    return a>b;
}

int main()
{
    int a[50], b[50], len, c, result = 0;
    std::cin >> len;
    for(int i = 0; i < len; i++)
    {
        std::cin >> c;
        a[i] = c;
    }
    for(int i = 0; i < len; i++)
    {
        std::cin >> c;
        b[i] = c;
    }
    std::sort(a, a+len);
    std::sort(b, b+len, calc);
    for(int i = 0; i < len; i++)
    {
        result += a[i]*b[i];
    }
    std::cout << result;
}

File: BOJ/Python/1026-보물.py
_ = input()
a = [int(i) for i in input().split()]
b = [int(i) for i in input().split()]
a.sort()
b.sort(reverse = True)
result = 0
for i in range(len(b)):
    result += a[i]*b[i]
print(result)